compiler/rustc_passes/src/check_attr.rs

```diff
@@ -388,10 +388,25 @@ impl CheckAttrVisitor<'tcx> {
             .emit();
     }
 
-    fn check_doc_alias(&self, meta: &NestedMetaItem, hir_id: HirId, target: Target) -> bool {
-        let doc_alias = meta.value_str().map(|s| s.to_string()).unwrap_or_else(String::new);
+    fn check_doc_alias_value(
+        &self,
+        meta: &NestedMetaItem,
+        doc_alias: &str,
+        hir_id: HirId,
+        target: Target,
+        is_list: bool,
+    ) -> bool {
         if doc_alias.is_empty() {
-            self.doc_attr_str_error(meta, "alias");
+            self.tcx
+                .sess
+                .struct_span_err(
+                    meta.name_value_literal_span().unwrap_or_else(|| meta.span()),
+                    &format!(
+                        "`#[doc(alias{})]` attribute cannot have empty value",
+                        if is_list { "(\"...\")" } else { " = \"...\"" },
+                    ),
+                )
+                .emit();
```

Comment:
`struct_` is redundant here.
```suggestion
                .span_err(
                    meta.name_value_literal_span().unwrap_or_else(|| meta.span()),
                    &format!(
                        "`#[doc(alias{})]` attribute cannot have empty value",
                        if is_list { "(\"...\")" } else { " = \"...\"" },
                    ),
                );
```